features/conversations/utils/search.ts

```diff
@@ -0,0 +1,122 @@
+import { getInboxProfileSocialsQueryData } from "@/queries/useInboxProfileSocialsQuery";
+import { getPreferredInboxName } from "@/utils/profile";
+import type {
+  DmWithCodecsType,
+  GroupWithCodecsType,
+} from "@/utils/xmtpRN/client";
+import type { InboxId } from "@xmtp/react-native-sdk";
+
+type DmSearchParams = {
+  account: string;
+  searchQuery: string;
+  dm: DmWithCodecsType;
+};
+
+export const dmMatchesSearchQuery = async ({
+  account,
+  searchQuery,
+  dm,
+}: DmSearchParams): Promise<boolean> => {
+  if (!dm) {
+    return false;
+  }
+  const inboxId = await dm.peerInboxId();
+  if (await inboxIdMatchesSearchQuery({ account, searchQuery, inboxId })) {
+    return true;
+  }
+  const members = await dm.members();
+  for (const member of members) {
+    if (
+      await addressMatchesSearchQuery({
+        searchQuery,
+        address: member.addresses[0],
+      })
+    ) {
+      return true;
+    }
+  }
+  return false;
+};
+
+type GroupSearchParams = {
+  account: string;
+  searchQuery: string;
+  group: GroupWithCodecsType;
+};
+
+export const groupMatchesSearchQuery = async ({
+  account,
+  searchQuery,
+  group,
+}: GroupSearchParams): Promise<boolean> => {
+  if (!group) {
+    return false;
+  }
+  if (group.name.toLowerCase().includes(searchQuery.toLowerCase())) {
+    return true;
+  }
+  const members = await group.members();
+  for (const member of members) {
+    if (
+      await inboxIdMatchesSearchQuery({
+        account,
+        searchQuery,
+        inboxId: member.inboxId,
+      })
+    ) {
+      return true;
+    }
+    if (
+      await addressMatchesSearchQuery({
+        searchQuery,
+        address: member.addresses[0],
+      })
+    ) {
+      return true;
+    }
+  }
+  return false;
+};
+
+type InboxIdSearchParams = {
+  account: string;
+  searchQuery: string;
+  inboxId: InboxId;
+};
+
+const inboxIdMatchesSearchQuery = async ({
+  account,
+  searchQuery,
+  inboxId,
+}: InboxIdSearchParams): Promise<boolean> => {
+  if (!inboxId) {
+    return false;
+  }
```

Comment:
Why do we have this if in the typing inboxId cannot be null or undefined

Reply:
You don't really lose anything from checking this and types are not final, but will remove